source/task_manager/crawl_tasks.py

```diff
@@ -70,39 +72,54 @@ def nutch(self, crawl, rounds=1, *args, **kwargs):
     job_client = nutch_client.Jobs(self.crawl.name)
 
     rest_crawl = nutch_client.Crawl(seed, jobClient=job_client, rounds=self.crawl.rounds_left)
+    nutch_crawl(self.crawl, rest_crawl, url_trails)
 
-    while self.crawl.rounds_left:
+
+def nutch_crawl(crawl_model, rest_crawl, url_trails):
+    """
+    Run a Nutch crawl until self.rounds_left is 0 or self.crawl.status is externally set to "STOPPING"
+
+    :param crawl_model: The crawl model
+    :param rest_crawl: A NutchPy Crawl Client
+    :param url_trails: A UrlTrails object for visualizing streaming messages from Nutch
+    """
+
+    while crawl_model.rounds_left:
         if rest_crawl.currentJob is None:
             rest_crawl.currentJob = rest_crawl.jobClient.create('GENERATE')
 
         active_job = rest_crawl.progress(nextRound=False)
         while active_job:
+            crawl_model = Crawl.objects.get(id=crawl_model.id)
+            sys.stderr.write("Crawler Status: {}".format(crawl_model.status))
+            if crawl_model.status == 'STOPPING':
+                crawl_model.status = 'STOPPED'
+                crawl_model.save()
```

Comment:
@brittainhard - the `crawl_mode.save()` call *must* be here.  Calling `save` from the outer function uses a stale reference with out of date information.